### ISSUE TITLE
Ip leaks

### ISSUE THREAD

**Issue by xakep2007** [projectdiscovery/httpx#432]
Any version.

```bash
$ echo 'http://ifconfig.me/all' | proxychains httpx -er '.*'
...
...via: 1.1 google,forwarded: YOU_REAL_IP ...
```

**Comment by ehsandeep** [comment]
@xakep2007 do you have more details to add about the issue/bug in httpx?

**Comment by xakep2007** [comment]
| prog | ver |
| --- | --- |
| proxychains-ng | 4.15 |
| torsocks | 2.3.0 |
| tor | 0.4.6.8. |
| httpx | v1.1.3 |

**Comment by ehsandeep** [comment]
@xakep2007 Yeah, that's the information revealed by the http://ifconfig.me/all website, when you connect to any website, your IP address is necessary, so that's expected; the point is, what does this have to do with httpx?

**Comment by xakep2007** [comment]
> @xakep2007 Yeah, that's the information revealed by the http://ifconfig.me/all website, when you connect to any website, your IP address is necessary, so that's expected; the point is, what does this have to do with httpx?

It's only httpx bug:

```bash
$ torify curl ifconfig.me                    
89.234.157.254
```

89.234.157.254 is Tor Exit Node.

**Comment by ehsandeep** [comment]
@xakep2007 can you share the output of `echo 'http://ifconfig.me' | torify httpx -unsafe -er '.*'`? i.e using `unsafe` flag with httpx
